### ISSUE TITLE
Tensor shape mismatches the associate numpy array shape

### ISSUE THREAD

**Issue by Evgeny-IT** [tensorflow/tensorflow#53451]
The following code demonstrates a simple TF graph with transpose, pad and another transpose that gives the output data with shape that mismatches tensor shape.

```
import numpy as np
import tensorflow.compat.v1 as tf

if __name__ == '__main__':
    tf.disable_eager_execution()
    tf.reset_default_graph()

    input_ = tf.placeholder(shape=(5, 3, 2, 4), dtype=tf.float32)

    permutation1 = tf.constant(value=(0, 2, 3, 1))
    transpose1 = tf.transpose(a=input_, perm=permutation1)

    pad_shape = tf.constant(value=((0, 0), (0, 0), (0, 0), (0, 0)))
    # pad_shape = tf.constant(value=((0, 0), (10, 0), (0, 0), (0, 0)))  # THIS VERSION WORKS!!!

    pad = tf.pad(tensor=transpose1, paddings=pad_shape)

    identity1 = tf.identity(input=pad)

    permutation2 = tf.constant(value=(0, 3, 1, 2))
    out_tensor = tf.transpose(a=identity1, perm=permutation2)

    tf.identity(out_tensor)

    input_data = np.random.randint(low=-127, high=127, size=np.prod(input_.shape)).reshape(input_.shape).astype(
        np.float32)

    with tf.Session() as session:
         tf_out_arrays = session.run(fetches=[out_tensor], feed_dict={input_: input_data})

    output_data = tf_out_arrays[0]

    assert input_.shape == input_data.shape, 'Input data shape'
    assert output_data.shape == out_tensor.shape, f'Output data shape mismatch {output_data.shape} != {out_tensor.shape}'
```

Gives the assertion error:
```
AssertionError: Output data shape mismatch (5, 4, 3, 2) != (5, 3, 2, 4)
```

The transposition was not applied by the TF and tensor shape now mismatches the data shape.

Interestingly if the padding values are not zeros, that is if the line
```
  pad_shape = tf.constant(value=((0, 0), (0, 0), (0, 0), (0, 0)))
```
is changed to 
```
pad_shape = tf.constant(value=((0, 0), (10, 0), (0, 0), (0, 0)))
```
the code works.

The TF version is:
```
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"
2021-12-16 11:45:20.516535: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2021-12-16 11:45:20.516563: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
v2.4.0-49-g85c8b2a817f 2.4.1
```
but I have also tested it against 2.4.4, 2.5.2, 2.6.2  and 2.7.0 - the mismatch reproduces. 

The code was tested in Ubuntu 20.04 and ArchLinux.

No GPU was used.

**Comment by tilakrayal** [comment]
@Saduf2019 ,
I was able to reproduce the issue in tf v2.7, v2.5 and nightly.Please find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/f6426f728b851140c5a515b19921be4a/53451.ipynb).

**Comment by Saduf2019** [comment]
@Evgeny-IT 
We see that you have not filled the issue template and used 1.x, the error reported is because the two shapes are not same.
As this is not bug or feat request hence please create this issue in discussion forum and move this to closed status.

**Comment by Evgeny-IT** [comment]
@Saduf2019

> We see that you have not filled the issue template

No, I filled them all. What fields of issue template do you think are missing?

> used 1.x,

No, I used 2.x, see the reply of @tilakrayal - he was able to reproduce the issue for 2.x package.

> As this is not bug

No, the error is here and is reproduced. Why this is a not bug? The expected behavior is not met.

**Comment by Saduf2019** [comment]
@Evgeny-IT 
As you can see you have used disable eager that means you are using tf 1.x, as explained this is not a bug we cannot fix coding issues in this repo, kindly open this in discussion forum.

**Comment by Evgeny-IT** [comment]
@Saduf2019 

> explained this is not a bug

Can you please give a more detailed explanation why this is not a bug? I see statements that this is not a bug and not explanation why. tf 1.x bugs are not considered bugs?

**Comment by Saduf2019** [comment]
@Evgeny-IT 
There is no support for tf 1.x anymore hence we cannot look into its issues. you are using disable and expecting it to work on 2.x this is incorrect coding.

**Comment by sushreebarsa** [comment]
@Evgeny-IT Session.run is deprecated. We see that you're using Session which does not work with  eager execution
 (TF V2). To migrate code that uses sessions to TF2, rewrite the code without it. Please see the [migration guide](https://www.tensorflow.org/guide/migrate#1_replace_v1sessionrun_calls) on replacing Session.run calls.

TF v1.x is currently  not actively supported so please try to upgrade to TF v2.4 or later versions.
For any further queries regarding TF v1.x specific issues please post it in [TF forum](https://discuss.tensorflow.org/) where there is a larger community to get you the right help.
Thanks!
